portalnet/src/storage.rs

```diff
@@ -744,6 +677,496 @@ impl PortalStorage {
+        let storage_usage = get_total_size_of_directory_in_bytes(&self.node_data_dir)?;
+        Ok(storage_usage)
+    }
+
+    /// Internal method for measuring the total amount of requestable data that the node is storing.
+    fn get_total_storage_usage_in_bytes_from_network(&self) -> Result<u64, ContentStoreError> {
+        let conn = self.sql_connection_pool.get()?;
+        let mut query = conn.prepare(TOTAL_DATA_SIZE_QUERY)?;
+
+        let result = query.query_map([], |row| {
+            Ok(DataSize {
+                num_bytes: row.get(0)?,
+            })
+        });
+
+        let sum = match result?.next() {
+            Some(total) => total,
+            None => {
+                let err = "Unable to compute sum over content item sizes".to_string();
+                return Err(ContentStoreError::Database(err));
+            }
+        }?
+        .num_bytes;
+
+        self.metrics.report_content_data_storage_bytes(sum);
+
+        Ok(sum as u64)
+    }
+
+    pub fn lookup_content_key(&self, id: [u8; 32]) -> anyhow::Result<Option<Vec<u8>>> {
+        let conn = self.sql_connection_pool.get()?;
+        let mut query = conn.prepare(CONTENT_KEY_LOOKUP_QUERY)?;
+        let id = id.to_vec();
+        let result: Result<Vec<BeaconContentKey>, ContentStoreError> = query
+            .query_map([id], |row| {
+                let row: String = row.get(0)?;
+                Ok(row)
+            })?
+            .map(|row| {
+                // value is stored without 0x prefix, so we must add it
+                let bytes: Vec<u8> = hex_decode(&format!("0x{}", row?))?;
+                BeaconContentKey::try_from(bytes).map_err(ContentStoreError::ContentKey)
+            })
+            .collect();
+
+        match result?.first() {
+            Some(val) => Ok(Some(val.into())),
+            None => Ok(None),
+        }
+    }
+
+    /// Public method for looking up a light client update by period number
+    pub fn lookup_lc_update_period(&self, period: u64) -> anyhow::Result<Option<u64>> {
+        let conn = self.sql_connection_pool.get()?;
+        let mut query = conn.prepare(LC_UPDATE_PERIOD_LOOKUP_QUERY)?;
+
+        let rows: Result<Vec<u64>, rusqlite::Error> = query
+            .query_map([period], |row| {
+                let row: u64 = row.get(0)?;
+                Ok(row)
+            })?
+            .collect();
+
+        match rows?.first() {
+            Some(val) => Ok(Some(*val)),
+            None => Ok(None),
+        }
+    }
+
+    /// Public method for looking up a content value by its content id
+    pub fn lookup_content_value(&self, id: [u8; 32]) -> anyhow::Result<Option<Vec<u8>>> {
+        let conn = self.sql_connection_pool.get()?;
+        lookup_content_value(id, conn)?
+    }
+
+    /// Public method for looking up a  light client update value by period number
+    pub fn lookup_lc_update_value(&self, period: u64) -> anyhow::Result<Option<Vec<u8>>> {
```

Comment:
Also, isn't this exactly the same as `lookup_lc_update_period()`?

Reply:
Similar logic as above, `lookup_lc_update_period` returns light client update period number while `lookup_lc_update_value` returns the light client update ssz bytes (content value).